executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -85,6 +87,11 @@ impl<'a, T: FieldElement> BlockMachineProcessor<'a, T> {
         }
     }
 
+    fn row_range(&self) -> std::ops::Range<i32> {
+        // We iterate over all rows of the block +/- one row, so that we can also solve for non-rectangular blocks.
+        -1..(self.block_size + 1) as i32
```

Comment:
I think at some point we should really try to auto-discover, i.e. as long as we make progress, we extend the range by one more. This could be problematic for fully-constant columns, though.

Reply:
Yeah, exactly. I think in practice this should be sufficient, we have the same assumption in the runtime solver.